### DISCUSSION TITLE
Can't make self.model.grid work in the agents

### DISCUSSION THREAD

**Discussion by NXabo** [mesa/mesa#1848] [Q&A]
Hello everyone, 

When trying to use the attribute grid (more specifically its get_cell_list_contents function) of the model used in the init method of an agent, python said that this attribute does not exist. The error is :

_**AttributeError: 'Pred_Agent' object has no attribute 'grid'**_

and the code :

animals_in_cell = self.model.grid.get_cell_list_contents([self.pos])

used in a function define in the Agent. This function is called by the step() function of the same Agent.

Can you help me so i can use this function in my modelisation ? 
To note : the code provided is from the tutorial found at :
https://mesa.readthedocs.io/en/main/tutorials/intro_tutorial.html

**Comment by rht** [comment]
That error means your `self.model` is not a `mesa.Model` object, but instead of a `Pred_Agent` agent object. You seemed to initialize it using the wrong class.

**Reply by NXabo** [reply]
I have found the error, it seems i was initializing agent giving it an agent instead of a model !

Thanks for your answer
